combat-trainer.lic

```diff
@@ -189,12 +190,47 @@ class LootProcess
   def dispose_body(game_state)
     return if DRRoom.dead_npcs.length < 1
     game_state['mob_died'] = true
-
-    check_skinning(DRRoom.dead_npcs.first) if @skin
-
+      if @necro
+        check_thanatology(DRRoom.dead_npcs.first)  #NECRO THANATOLOGY CHECK STUFF 
+      elseif @skin
+        check_skinning(DRRoom.dead_npcs.first)
+      end
+    dispose_body_loot
     bput('loot', 'You search', 'I could not find what you were referring to')
     pause 0.2 while DRRoom.dead_npcs.length > 0
   end
+  
+  def check_thanatology(mob_noun)  #NECRO THANATOLOGY RITUAL METHOD 
+    return if @construct.include?(mob_noun)
+    perform_type = @perform_types[mob_noun] || 'somethinggoeshere' #Setting this up like the skinning method because I have no idea what I'm doing.
+    perform_message = "perform #{perform_type} on #{mob_noun}"
+      bput (perform_message, 'Rituals do not work upon constructs', 'corpse to make one long,', 'carefully position the corpse', 'corpse and make a few quick precise cuts', 'corpse to make a few quick,')
+      when 'corpse to make one long,', 'carefully position the corpse', 'corpse and make a few quick precise cuts', 'corpse to make a few quick,'
+        break
```

Comment:
check_skinning at line 236 has a when for each condition , is there a reason it does?